include/index.h

```diff
@@ -192,11 +192,11 @@ namespace diskann {
     // Initialize space for res_vectors before calling.
     DISKANN_DLLEXPORT size_t search_with_tags(const T *query, const uint64_t K,
                                               const unsigned L, TagT *tags,
-                                              float *           distances,
+                                              float            *distances,
                                               std::vector<T *> &res_vectors);
 
-    // Will fail if tag already in the index
-    DISKANN_DLLEXPORT int insert_point(const T *point, const TagT tag);
+    // Will fail if tag already in the index or if tag=0.
+    DISKANN_DLLEXPORT int insert_point(const T *point, const TagT &tag);
```

Comment:
I think we're limiting `TagT` to primitive types (there are asserts to that effect in e.g. `natural_number_map.h` and `index` template is instantiated only for a few primitive types). Given that, reference for tag just adds unnecessary level of indirection.

Reply:
I can remove the reference for now. Is it easy to generalize the nnmap.h to string tag types?

Reply:
AFAIR the main thing would be to make sure that primitive types are passed by value and `std::string` etc. by reference. Should be relatively straightforward, but I didn't want to spent time on it when creating the class as it wasn't necessary.